### PR TITLE
[explorer] Reinstate test suite, fix latency issues and split view/fetch logic

### PR DESCRIPTION
**Pull request by Andrew47**
This PR adds the latest changes I've been working on. It will be updated to reflect the latest changes to the `explorer-merge` branch.

### CODE REVIEW

**Comment by Andrew47** [comment]
Found and Resolved following bugs:

* Dog disappears when Owner is clicked:  https://explorer-demo.sui.io/objects/363124FB1BE19377C39C4356865A8C8430C6AE85
*  Image stuck on 'Please Wait': https://explorer-demo.sui.io/objects/A033A5126D3EC043A5CF259075B9F7A4F4E789EB

**Comment by Andrew47** [comment]
@stella3d --> How is `./explorer/client/src/utils/demoTypes.ts` used? I could not find another file that imports this file. Can we remove this file?

**Review by stella3d — CHANGES_REQUESTED** [review]
change look good but they don't play nice with the main branch yet - i tried doing a rebase but it got messy.

One we get the merge conflicts sorted out this should be good

**Comment by Andrew47** [comment]
@stella3d --> I've rebased this branch on main and resolved the merge conflicts.

File: explorer/client/README.md [hunk 1]
```diff
@@ -10,6 +10,26 @@ In the project directory, run:
...
+What the 'Sui Network' is varies according to the environment variable `REACT_APP_DATA`.
+
+When running most of the below npm commands, the SuiExplorer Client will extract and present data from the Sui Network connected to the URL https://demo-rpc.sui.io.
+
+If the environment variable `REACT_APP_DATA` is set to `static`, then the SuiExplorer will instead pull data from a local, static JSON dataset that can be found at `./src/utils/static/mock_data.json`.
+
+For example, suppose we wish to locally run the website using the static JSON dataset and not the API, then we would run the following:
+
+```bash
+$ REACT_APP_DATA=static npm start
```
Comment (Clay-Mysten) [inline explorer/client/README.md:26]: I would remove the dollar sign as it gets picked up by the auto-copy functions of both GitHub and our Dev Portal. We include it only when output is also included in the same code block. (I recommend splitting those out anyway.)

**Review by Clay-Mysten — APPROVED** [review]
Just one optional suggestion. Thanks, Andrew!

File: explorer/client/README.md (hunk 1 above)
Comment (Andrew47) [reply explorer/client/README.md]: Thanks Clay. I've removed the dollar sign.

**Review by Clay-Mysten — APPROVED** [review]
Thanks, Andrew!

**Review by stella3d — APPROVED** [review]

**Pull request merged**